test/benchgc/src/benchgc/tuner/op_config.py

```diff
@@ -0,0 +1,175 @@
+        self.M_threads = M_threads
+        self.K_threads = K_threads
+        self.N_threads = N_threads
+        self.M_block = M_block
+        self.K_block = K_block
+        self.N_block = N_block
+        self.innermostM_block = innermostM_block
+        self.innermostK_block = innermostK_block
+        self.innermostN_block = innermostN_block
+        super().__init__()
+
+    def init_candidates(self):
+        default_blocks = [16, 32, 64, 128, 256, 512]
+        default_innermost_blocks = [16, 32]
+        self.field_candidates["M_threads"] = find_factors(self.num_threads)
+        self.field_candidates["K_threads"] = find_factors(self.num_threads)
+        self.field_candidates["N_threads"] = find_factors(self.num_threads)
+        self.field_candidates["M_block"] = [
+            block for block in default_blocks if self.M >= block
+        ]
+        self.field_candidates["K_block"] = [
+            block for block in default_blocks if self.K >= block
+        ]
+        self.field_candidates["N_block"] = [
+            block for block in default_blocks if self.N >= block
+        ]
+        self.field_candidates["innermostM_block"] = [
+            block for block in default_innermost_blocks if self.M >= block
+        ]
+        self.field_candidates["innermostK_block"] = [
+            block for block in default_innermost_blocks if self.K >= block
+        ]
+        self.field_candidates["innermostN_block"] = [
+            block for block in default_innermost_blocks if self.N >= block
+        ]
+
+    def init_constraints(self):
+        # example: using lambda to add constraints, adding constraints by the order of the fields
+        self.field_constraints["M_threads"] = None
+        self.field_constraints["K_threads"] = (
+            lambda MatMulConfig, K_threads: self.num_threads
+            % (MatMulConfig.M_threads * K_threads)
+            == 0
+        )
+        self.field_constraints["N_threads"] = (
+            lambda MatMulConfig, N_threads: self.num_threads
+            % (MatMulConfig.M_threads * MatMulConfig.K_threads * N_threads)
+            == 0
+        )
+        self.field_constraints["M_block"] = None
+        self.field_constraints["K_block"] = None
+        self.field_constraints["N_block"] = None
+        self.field_constraints["innermostM_block"] = (
+            lambda MatMulConfig, innermostM_block: MatMulConfig.M_block
+            % innermostM_block
+            == 0
+        )
+        self.field_constraints["innermostK_block"] = (
+            lambda MatMulConfig, innermostK_block: MatMulConfig.K_block
+            % innermostK_block
+            == 0
+        )
+        self.field_constraints["innermostN_block"] = (
+            lambda MatMulConfig, innermostN_block: MatMulConfig.N_block
+            % innermostN_block
+            == 0
+        )
+
+    def attach_to_ir(self, op: OpView):
+        attr_to_field = {
+            "Mthreads": self.M_threads,
```

Comment:
Currently `MatmulConfigAnalysis.cpp` reads named attribute `MThreads` instead of `Mthreads`. Please align the naming convention here (also for `Kthreads` and `Nthreads`).